### PR TITLE
feat: Scrub all images and other various media

### LINKED ISSUE #232: Scrub images by default
**Problem:**

Today, our default privacy configuration masks texts and inputs on replay videos. However, images still are present, potentially leaking sensitive information.

**Solution:**

- [ ] Scrub/mask images by default on user sessions
- [ ] Update the [README.md](https://github.com/getsentry/sentry-replay#sentry-replay) to reflect the changes

### PR DESCRIPTION
**Pull request by billyvg**
Add a new option `blockAllMedia` (default: true) that will "block" various media elements.

Closes #232

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,6 +64,7 @@ replay.stop(); // Stop recording
 | key              | type                     | default                             | description                                                                                                                                                                                         |
 | ---------------- | ------------------------ | ----------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | maskAllText      | boolean                  | `true`                              | Mask _all_ text content. Will pass text content through `maskTextFn` before sending to server.                                                                                                       |
+| blockAllMedia    | boolean                  | `true`                              | Block _all_ media elements (`img, svg, video, object, picture, embed, map, audio`)
```
Comment (billyvg) [inline README.md:67]: svg too?

**Comment by billyvg** [comment]
here's what this looks like in sentry: https://sentry.io/organizations/sentry-emerging-tech/replays/billy-test:a2fc0b8ae74b4c1180b7cb2b10bb0099/?referrer=%2Forganizations%2F%3AorgId%2Freplays%2F

File: README.md [hunk 2]
```diff
@@ -84,7 +85,7 @@ replay.stop(); // Stop recording
...
 
 ## Privacy
-There are several ways to deal with PII. By default, the integration will mask all text content with `*`. This can be disabled by setting `maskAllText` to `false`. It is also possible to add the following CSS classes to specific DOM elements to prevent recording its contents: `sentry-block`, `sentry-ignore`, and `sentry-mask`. The following sections will show examples of how content is handled by the differing methods.
+There are several ways to deal with PII. By default, the integration will mask all text content with `*` and block all media elements (`img, svg, video, object, picture, embed, map, audio`). This can be disabled by setting `maskAllText` to `false`. It is also possible to add the following CSS classes to specific DOM elements to prevent recording its contents: `sentry-block`, `sentry-ignore`, and `sentry-mask`. The following sections will show examples of how content is handled by the differing methods.
```
Comment (JoshFerge) [inline README.md:88]: what about: 

canvas, 
iframe?

**Review by JoshFerge — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (billyvg) [reply README.md:88]: canvas has an explicit `recordCanvas` option which is default `false`

I'll look into iframe

File: src/index.ts [hunk 3]
```diff
@@ -59,6 +59,8 @@ type AddUpdateCallback = () => boolean | void;
 const BASE_RETRY_INTERVAL = 5000;
 const MAX_RETRY_COUNT = 3;
 const UNABLE_TO_SEND_REPLAY = 'Unable to send Replay';
+const MEDIA_SELECTORS =
+  'img,svg,path,rect,area,video,object,picture,embed,map,audio';
```
Comment (ryan953) [inline src/index.ts:63]: i think we should add `image` to the list, it's not valid but there are carveouts in the spec for it to be converted to `img`

File: README.md (hunk 2 above)
Comment (billyvg) [reply README.md:88]: @JoshFerge [rrweb strips the iframe src](https://github.com/rrweb-io/rrweb/blob/6f44bb72a194ad428bc697229f589c17d713da2c/packages/rrweb-snapshot/src/snapshot.ts#L776-L783)

image

Comment (JoshFerge) [reply README.md:88]: i thought on our replay details page we could see the replay playing in the replay? how was that working? does that not use an iframe?

Comment (billyvg) [reply README.md:88]: It does but the player manipulates the contents of the iframe rather than embedding external content within it

**Pull request merged**